Sources/PrimerSDK/Classes/Extensions & Utilities/StringExtension.swift

```diff
@@ -130,7 +152,18 @@ internal extension String {
             return count == numberOfDigits
         }
         
-        return count > 2 && count < 5
+        let isValid = count > 2 && count < 5
+        
+        if !isValid {
+            let event = Analytics.Event(
```

Comment:
actually, would it maybe be better to call the analytics service inside view models & other services? It feels a bit heavy to put all of this inside utility methods

Reply:
You're right, fixed.